src/configurationProvider.ts

```diff
@@ -189,6 +189,44 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
     private isEmptyArray(configItems: any): boolean {
         return !Array.isArray(configItems) || !configItems.length;
     }
+
+    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption | undefined> {
+        const res = await resolveMainClass(folder ? folder.uri : undefined);
+        if (res.length === 0) {
+            vscode.window.showErrorMessage(
+                "Cannot find a class with the main method.");
+            return undefined;
+        }
+        const pickItems = res.map((item) => {
+            let name = item.mainClass;
+            let details = `main class: ${item.mainClass}`;
+            if (item.projectName !== undefined) {
+                name += `<${item.projectName}>`;
+                details += ` | project name: ${item.projectName}`;
+            }
+            return {
+                description: details,
+                label: name,
+                item,
+            };
+        }).sort ((a, b): number => {
+            return a.label > b.label ? 1 : -1;
+        });
+        if (pickItems.length === 0) {
+            // user gives up the selection, quit sliently
```

Comment:
There is no items. User doesn't have the selection yet. Should remove this comment?